oracle/provider/binance.go

```diff
@@ -125,6 +125,25 @@ func NewBinanceProvider(
 		subscribedPairs: map[string]types.CurrencyPair{},
 	}
 
+	availablePairs, err := provider.GetAvailablePairs()
+	if err != nil {
+		return nil, err
+	}
+
+	// confirm pairs can be subscribed to
+	for i, pair := range pairs {
+		if _, ok := availablePairs[pair.String()]; ok {
```

Comment:
Is this something we want to check for on `SubscribeCurrencyPairs` as well?

Reply:
I added a new provider method `ConfirmPairAvailability` to do this check and added that check to `SubscribeCurrencyPairs`